emily_cron/app/services/deposit_processor.py

```diff
@@ -0,0 +1,191 @@
+
+from ..clients import PrivateEmilyAPI, HiroAPI, MempoolAPI
+from ..models import (
+    DepositUpdate,
+    EnrichedDepositInfo,
+    RequestStatus,
+    DepositInfo,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class DepositProcessor:
+    """Service for processing deposits."""
+
+    def process_expired_locktime(
+        self,
+        enriched_deposits: list[EnrichedDepositInfo],
+        bitcoin_chaintip_height: int,
+    ) -> list[DepositUpdate]:
+        """Process transactions with expired locktime.
+
+        This includes:
+        - Transactions where locktime expired and the UTXO was NOT spent.
+        - Transactions where the UTXO was reclaimed.
+
+        Args:
+            enriched_deposits: List of enriched deposit information
+            bitcoin_chaintip_height: Current Bitcoin chain tip height
+
+        Returns:
+            list[DepositUpdate]: List of deposit updates
+        """
+        updates = []
+
+        for tx in enriched_deposits:
+            # Step 1: Check if the time-based expiry condition is met
+            if not tx.is_expired(bitcoin_chaintip_height):
+                continue  # Not eligible for failure yet based on time
+
+            # Step 2: Time has expired, now check UTXO status
+            logger.debug(f"Deposit {tx.bitcoin_txid} time expired, checking UTXO status...")
+            utxo_status = MempoolAPI.get_utxo_status(tx.bitcoin_txid, tx.bitcoin_tx_output_index)
+            is_utxo_spent = utxo_status.get("spent", False)
+
+            if not is_utxo_spent:
+                # Case 1: Time expired AND UTXO is unspent -> Mark FAILED
+                logger.info(
+                    f"Marking transaction {tx.bitcoin_txid} as FAILED (Locktime expired, unspent)"
+                )
+                updates.append(
+                    DepositUpdate(
+                        bitcoin_txid=tx.bitcoin_txid,
+                        bitcoin_tx_output_index=tx.bitcoin_tx_output_index,
+                        status=RequestStatus.FAILED.value,
+                        status_message=f"Locktime expired at height {bitcoin_chaintip_height} and UTXO unspent",
+                    )
+                )
+                continue
+
+            # Step 3: UTXO is spent, check if it was a reclaim
+            spending_txid = utxo_status.get("txid")
+            if not spending_txid:
+                logger.warning(
+                    f"Deposit {tx.bitcoin_txid} UTXO is spent, but spending TXID is missing from API response. Cannot check for reclaim."
+                )
+                continue  # Cannot determine reclaim, assume it's a signer sweep
+
+            logger.debug(
+                f"Deposit {tx.bitcoin_txid} UTXO spent by {spending_txid}, checking for reclaim..."
+            )
+            spending_tx_details = MempoolAPI.get_transaction(spending_txid)
+            is_reclaim_check = False
+            if spending_tx_details:
+                for vin in spending_tx_details.get("vin", []):
```

Comment:
nit: the `outspend` API returns `vin` (index), we could use it to avoid finding it again